### PR TITLE
Data: Add Airtable API Option

### PR DESCRIPTION
**Pull request by gblakeman**
Adds the ability to use our data directly from Airtable with the repo-committed data as a fallback. If generated (Airtable) data is found, the plugin will `run`/`build` with that data, otherwise it will use the data from the repo.

To set up, add and `AIRTABLE_API_KEY` and `AIRTABLE_BASE_ID` to the environments file. You will need access to the appropriate Airtable base.

To update data, run the command: `npm run data-update`.

------

To-do:
- [x] Docs
- [x] README updates

------
Supersedes #62 and removes unnecessary test work.

### CODE REVIEW

**Review by lynnerang — APPROVED** [review]
Awesommmme!!

File: package.json [hunk 1]
```diff
@@ -22,19 +22,21 @@
     "start:public": "npm run watch:public",
     "start:internal": "npm run watch:internal",
     "start:beta": "npm run watch:beta",
+    "update": "npm run data-update",
```
Comment (joeyyang) [inline package.json:25]: nit: `update` seems a bit too general - i'd be wondering "updating what (specifically)?" - perhaps just only have `npm run data-update` as the data update command?

File: src/data/run-data-update.mjs [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+            }
+          }
+        }
+
+        // set string for array in file
+        if (formatImageCheck && hasImage) {
+          const string = `  {\n    name: '${item.fields.Text}',\n    hasImage: ${hasImage},\n  },`;
+          itemBundleArray.push(string);
+        } else {
+          itemBundleArray.push(`  '${item.fields.Text}',`);
```
Comment (joeyyang) [inline src/data/run-data-update.mjs:144]: curious - why do we push the string with whitespace in front?

File: src/data/run-data-update.mjs [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+      route: url,
+    };
+    await makeNetworkRequest(payload, await extractDataRecords);
+
+    // ----- if a valid assignment match is found and data is retrieved, write the data file
+    if (currentAssignment) {
+      if (categoryDataRecords) {
+        const hasImages = currentAssignment.fields['Has image?'] === 'Yes';
+        const pluralName = pluralize(currentAssignment.fields['Data Key']).replace('companyMedias', 'companiesMedia');
+        const content = `/**\n * ${currentAssignment.fields['Data Type']}.\n *\n * @kind constant\n * @name ${pluralName}\n * @type {Array}\n */\nexport default [\n${formatArrayItems(categoryDataRecords, hasImages)}\n];`;
```
Comment (joeyyang) [inline src/data/run-data-update.mjs:168]: any way of breaking this up a bit?

File: src/data/run-data-update.mjs [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+    );
+
+    // add `unassigned` to the front as the default
+    const assignmentBundle = {
+      unassigned: { id: 'unassigned', text: 'Unassigned', nodeType: null },
+      ...orderedAssignmentsBundle,
+    };
+
+    // ------ write to file
+    const content = `/**\n * An object containing the available data type assignments. The \`id\` is\n * used for matching. The \`text\` is only used as a label. Change only the\n * \`text\` if you wish to maintain backwards compatibility to an existing\n * assignment.\n *\n * @kind constant\n * @name assignments\n * @type {Object}\n */\nexport default ${JSON.stringify(assignmentBundle, null, 2).replace(/"/g, '\'').replace(/(')+(\w+)+(':)+/g, '$2:')};\n`;
```
Comment (joeyyang) [inline src/data/run-data-update.mjs:289]: any way of breaking this string up a bit?

**Review by joeyyang — APPROVED** [review]
Looks good! A few readability and naming nits, but otherwise this is awesome!

File: src/data/run-data-update.mjs (hunk 2 above)
Comment (gblakeman) [reply src/data/run-data-update.mjs:144]: To create a human-readable javascript file. Keeping them readable helps me a lot when I’m making sure everything is coming in from Airtable correctly.

File: src/data/run-data-update.mjs (hunk 4 above)
Comment (gblakeman) [reply src/data/run-data-update.mjs]: I separated out the headers, but can’t do much else without affecting the whitespace that is inserted in the generated file.

**Pull request merged**